core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1126,39 +1127,47 @@ class DAGScheduler(
       case FetchFailed(bmAddress, shuffleId, mapId, reduceId, failureMessage) =>
         val failedStage = stageIdToStage(task.stageId)
         val mapStage = shuffleToMapStage(shuffleId)
+        if (failedStage.attemptId - 1 > task.stageAttemptId) {
```

Comment:
What's the intuition here  (I was expecting "if(failedStage.attemptId != task.stageAttemptId)")?